SimCalorimetry/EcalEBTrigPrimProducers/plugins/EcalEBTrigPrimPhase2Producer.cc

```diff
@@ -0,0 +1,185 @@
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+#include "CondFormats/EcalObjects/interface/EcalLiteDTUPedestals.h"
+#include "CondFormats/DataRecord/interface/EcalLiteDTUPedestalsRcd.h"
+
+// We keep these lines for future posssible necessary additions
+//#include "CondFormats/EcalObjects/interface/EcalTPGTowerStatus.h"
+//#include "CondFormats/DataRecord/interface/EcalTPGStripStatusRcd.h"
+//#include "CondFormats/EcalObjects/interface/EcalTPGStripStatus.h"
+
+#include "EcalEBTrigPrimPhase2Producer.h"
+
+#include <memory>
+
+#include "SimCalorimetry/EcalEBTrigPrimAlgos/interface/EcalEBPhase2TrigPrimAlgo.h"
+
+EcalEBTrigPrimPhase2Producer::EcalEBTrigPrimPhase2Producer(const edm::ParameterSet& iConfig)
+    : debug_(iConfig.getParameter<bool>("Debug")),
+      famos_(iConfig.getParameter<bool>("Famos")),
+      binOfMaximum_(iConfig.getParameter<int>("binOfMaximum")) {
+  tokenEBdigi_ = consumes<EBDigiCollectionPh2>(iConfig.getParameter<edm::InputTag>("barrelEcalDigis"));
+
+  eTTmapToken_ = esConsumes<edm::Transition::BeginRun>();
+  theGeometryToken_ = esConsumes<edm::Transition::BeginRun>();
+
+  theEcalTPGPedestals_Token_ =
+      esConsumes<EcalLiteDTUPedestalsMap, EcalLiteDTUPedestalsRcd, edm::Transition::BeginRun>();
+  theEcalEBPhase2TPGPedestals_Token_ =
+      esConsumes<EcalEBPhase2TPGPedestalsMap, EcalEBPhase2TPGPedestalsRcd, edm::Transition::BeginRun>();
+
+  theEcalTPGCrystalStatus_Token_ =
+      esConsumes<EcalTPGCrystalStatus, EcalTPGCrystalStatusRcd, edm::Transition::BeginRun>();
+  theEcalEBPhase2TPGLinearization_Token_ =
+      esConsumes<EcalEBPhase2TPGLinearizationConst, EcalEBPhase2TPGLinearizationConstRcd, edm::Transition::BeginRun>();
+  theEcalEBTPGAmplWeightIdMap_Token_ =
+      esConsumes<EcalEBPhase2TPGAmplWeightIdMap, EcalEBPhase2TPGAmplWeightIdMapRcd, edm::Transition::BeginRun>();
+  theEcalEBTPGTimeWeightIdMap_Token_ =
+      esConsumes<EcalEBPhase2TPGTimeWeightIdMap, EcalEBPhase2TPGTimeWeightIdMapRcd, edm::Transition::BeginRun>();
+  theEcalTPGWeightGroup_Token_ = esConsumes<EcalTPGWeightGroup, EcalTPGWeightGroupRcd, edm::Transition::BeginRun>();
+
+  //register your products
+  produces<EcalEBPhase2TrigPrimDigiCollection>();
+}
+
+void EcalEBTrigPrimPhase2Producer::beginRun(edm::Run const& run, edm::EventSetup const& setup) {
+  auto const& theGeometry = setup.getData(theGeometryToken_);
+  auto const& eTTmap = setup.getData(eTTmapToken_);
+
+  algo_ = std::make_unique<EcalEBPhase2TrigPrimAlgo>(&eTTmap, &theGeometry, binOfMaximum_, debug_);
+
+  // get a first version of the records
+  cacheID_ = this->getRecords(setup);
+
+  nEvent_ = 0;
+}
+
+void EcalEBTrigPrimPhase2Producer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<bool>("Debug", false);
+  desc.add<bool>("Famos", false);
+  desc.add<int>("BinOfMaximum", 6);  // this needs to be at the same value used for the Phase2 LiteDTU digis !
```

Comment:
This could be achieved by default with a constant in [EcalConstants.h](https://github.com/cms-sw/cmssw/blob/master/DataFormats/EcalDigi/interface/EcalConstants.h) that is used here and where it is needed for the LiteDTU digis.
The comment is still useful.